### PR TITLE
[CL: Audit] Superfluid Migration audit / cleanup

### PR DESCRIPTION
**Pull request by stackman27**
Closes: #XXX
Part of: https://github.com/osmosis-labs/osmosis/issues/5580

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/superfluid/keeper/migrate.go [hunk 1]
```diff
@@ -81,7 +95,8 @@ func (k Keeper) migrateSuperfluidBondedBalancerToConcentrated(ctx sdk.Context,
 	intermediateAccount := types.SuperfluidIntermediaryAccount{}
 	var gammLockToMigrate *lockuptypes.PeriodLock
 	if isPartialMigration {
-		// Note that lock's id is different from the originalLockId sinec it was split.
+
```
Comment (czarcas7ic) [inline x/superfluid/keeper/migrate.go:98]:
```suggestion
```

**Review by czarcas7ic — APPROVED** [review]
Thanks 👍

File: x/superfluid/keeper/migrate.go [hunk 2]
```diff
@@ -24,9 +24,23 @@ const (
 )
 
 // RouteLockedBalancerToConcentratedMigration routes the provided lock to the proper migration function based on the lock status.
-// If the lock is superfluid delegated, it will instantly undelegate the superfluid position and redelegate it as a concentrated liquidity position.
-// If the lock is superfluid undelegating, it will instantly undelegate the superfluid position and redelegate it as a concentrated liquidity position, but continue to unlock where it left off.
-// If the lock is locked or unlocking but not superfluid delegated/undelegating, it will migrate the position and either start unlocking or continue unlocking where it left off.
+//
```
Comment (mattverse) [inline x/superfluid/keeper/migrate.go:27]:
```suggestion
// The testing conditions and scope for the different lock status are as follows: 
```

**Review by mattverse — APPROVED** [review]

File: x/superfluid/keeper/migrate.go [hunk 3]
```diff
@@ -24,9 +24,23 @@ const (
...
+// Lock Status = Superfluid delegated
+// - Instantly undelegate which will bypass unbonding time.
+// - Create new CL Lock and Re-delegate it as a concentrated liquidity position.
+//
+// Lock Status = Superfluid undelegating
+// - Continue undelegating as superfluid unbonding CL Position.
+// - Lock the tokens and create an unlocking syntheticLock (to handle cases of slashing)
```
[the remark this answers is not in the document]
Comment (stackman27) [reply x/superfluid/keeper/migrate.go:34]: wouldn't that be this case? 
https://github.com/osmosis-labs/osmosis/pull/5615/files#diff-95611a6b4b0a087202f772a2bfa9dd21b37f06e24126d793d176d211ce56dbecR36-R38

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, thanks for cleaning this up!

**Comment by stackman27** [comment]
merging this since it has 3 approvals

**Pull request merged**